src/source/mod.rs

```diff
@@ -449,12 +449,12 @@ pub enum NpmPackageReqResolution {
   /// cached/loaded registry information and deno_graph will request
   /// the registry information for every package again then re-attempt resolution.
   ///
-  /// deno_graph will restart only once per build call to prevent accidental,
-  /// infinite loops, but the implementation should ensure this is only
-  /// returned once per session.
-  ReloadRegistryInfo(anyhow::Error),
+  /// The implementation MUST ensure this doesn't get returned more than
+  /// once in a row or else a panic will occur.
+  ReloadRegistryInfo,
```

Comment:
I debated completely removing this functionality from deno_graph, but I think it's better to just leave it here to get an implementor to think about it.